### PR TITLE
Add a legacy documentation section

### PR DESCRIPTION
**Pull request by didier-wenzek**
## Proposed changes

- Remove deprecated mdbook settings
- Add Legacy section
- Move child_device_config_management under legacy
- Move c8y-firmare-management docs under Legacy section

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/legacy/child_device_config_management_agent.md [hunk 1]
```diff
@@ -225,7 +225,6 @@ a "failed" status update must be sent instead, to the same topic as follows:
 
 ## References
 
-* Configuration Management [documentation](config_management_plugin.md)
 * Reference implementation of a [child device agent](https://github.com/thin-edge/thin-edge.io_examples/tree/main/child-device-agent) written in Python to demonstrate the contract described in this document.
```
Comment (reubenmiller) [inline docs/src/legacy/child_device_config_management_agent.md:230]: We can get rid of this section as we'll be updating the reference child device agent to use the new v1 syntax, so it won't match the contract described on this page anyway.

File: docs/src/legacy/child_device_config_management_agent.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ to the URL: `http://{fts-address}:8000/tedge/file-transfer/{child-id}/c8y-config
 * {fts-address}` is the address of the thin-edge.io device on which the [File Transfer Service][fts] is running
 * `{child-id}` is the child-device-id
 
-[fts]: ../../references/tedge-file-transfer-service.md
+[fts]: ../references/tedge-file-transfer-service.md
```
Comment (reubenmiller) [inline docs/src/legacy/child_device_config_management_agent.md:75]: I think we can delete this line as it is invalid markdown (to my knowledge) and it is not rendered on the page anyway...

File: docs/src/legacy/child_device_config_management_agent.md (hunk 1 above)
Comment (didier-wenzek) [reply docs/src/legacy/child_device_config_management_agent.md]: removed

File: docs/src/legacy/child_device_config_management_agent.md (hunk 2 above)
Comment (didier-wenzek) [reply docs/src/legacy/child_device_config_management_agent.md]: This is a so called [reference link](https://daringfireball.net/projects/markdown/syntax#link).

As this is unusual, I removed it.

File: docs/src/legacy/child_device_config_management_agent.md [hunk 3]
```diff
@@ -72,7 +72,7 @@ to the URL: `http://{fts-address}:8000/tedge/file-transfer/{child-id}/c8y-config
 * {fts-address}` is the address of the thin-edge.io device on which the [File Transfer Service][fts] is running
```
[the remark this answers is not in the document]
Comment (didier-wenzek) [reply docs/src/legacy/child_device_config_management_agent.md]: fixed

**Review by reubenmiller — APPROVED** [review]
Approved

**Pull request merged**